packages/client-node/src/connection/node_base_connection.ts

```diff
@@ -108,15 +108,21 @@ export abstract class NodeBaseConnection
   }
 
   async ping(): Promise<ConnPingResult> {
+    const query_id = this.getQueryId(undefined)
     const abortController = new AbortController()
     try {
+      const searchParams = toSearchParams({
+        database: undefined,
+        query: PingQuery,
+        query_id,
+      })
       const { stream } = await this.request(
         {
           method: 'GET',
-          url: transformUrl({ url: this.params.url, pathname: '/ping' }),
+          url: transformUrl({ url: this.params.url, searchParams }),
+          query: PingQuery,
```

Comment:
So, it's a breaking change since now ping validates the credentials? Can we add it via a flag as you proposed earlier? We can remove the flag in a major release. let's start tracking these breaking changes somewhere

Reply:
@mshustov, should be fixed